crates/primitives/src/transaction/envelope.rs

```diff
@@ -710,92 +639,31 @@ mod tests {
     }
```

Comment:
Looks like a handful of tests (`test_payment_classification_with_tip20_prefix`, `test_payment_classification_without_tip20_prefix`, `test_payment_classification_no_to_address`, `test_payment_classification_partial_match`, `test_payment_classification_different_prefix`) were removed since they used the `FeeToken` tx type. However these tests are stil useful, we should reintroduce these and update the tx envelope to use a different variant instead.